src/pipeline.ts

```diff
@@ -87,6 +86,26 @@ function toPipeline(steps: Step[]): Pipeline {
   return { env: {}, steps };
 }
 
+/**
+ * Mutates the config objects within ConfigWithDecision to account for transitive dependencies between pipelines
+ */
+function updateDecisionsForDependsOn(configs: ConfigWithDecision[]): void {
+  const byName = Object.fromEntries(configs.map((c) => [c.name, c]));
+
+  configs
+    .flatMap((config) => config.monorepo.depends_on.map((dependency) => [config.name, dependency]))
+    .reverse()
```

Comment:
The pipeline has already been ordered according to depends_on at this point. By reversing, we are processing depends_on from the end of the pipeline toward the front. That means even if there is a chain of dependencies being pulled in, we only need to iterate this list once and we'll get them all as long as we mark as we go.

That's my reasoning anyway...